<launch>
  <arg name="voxblox_path" default="/root/catkin_ws/src/mav_voxblox_planning/voxblox_sst_planner/maps/machine_hall/voxblox/rs_esdf_0.10.voxblox"  />
  <arg name="frame_id" default="map" />

  <node name="voxblox_sst_planner" pkg="voxblox_sst_planner" type="voxblox_sst_planner_node" output="screen">
    <param name="voxblox_path" value="$(arg voxblox_path)" />
    <param name="visualize" value="true" />
    <param name="robot_radius" value="0.25" />
    <param name="tsdf_voxel_size" value="0.10" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="update_mesh_every_n_sec" value="0.0" />
    <param name="num_seconds_to_plan" value="10.0" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="world_frame" value="$(arg frame_id)" />
    <param name="trust_approx_solution" value="true" />
    <param name="publish_traversable" value="true" />
    <param name="split_at_collisions" value="true" />
    <param name="optimize_time" value="true" />
    <param name="verbose" value="false" />
    <param name="do_smoothing" value="false"/>


    <param name="minControlDuration" value="1"/>
    <param name="maxControlDuration" value="50"/>
    <param name="stepSize" value="0.02"/>
    <param name="goalRadius" value="2.5"/>

    <param name="minRotateControl" value="-1.0"/>
    <param name="maxRotateControl" value="1.0"/>
    <param name="minZControl" value="-15.0"/>
    <param name="maxZControl" value="-5.0"/>
    <param name="minVel" value="-1.0"/>
    <param name="maxVel" value="1.0"/>
    <param name="minOmega" value="-1.0"/>
    <param name="maxOmega" value="1.0"/>

    <param name="savePath" value="true"/>
    <param name="saveDir" value="/root/catkin_ws/data"/>

  </node>

 <node type="rviz" name="rviz" pkg="rviz" />
</launch>
